### DISCUSSION TITLE
How can we make GPT aware of existing files in the "workspace" directory?

### DISCUSSION THREAD

**Discussion by vikito755** [AntonOsika/gpt-engineer#573] [Q&A]
I tried, typing out the directory structure, providing and used prompt engineering methods to order words (I assume) in a better way. Is there an existing way for this?

**Comment by patruff** [comment]
I think you should be able to use the Python REPL tool and then use os to run an LS command

I've done this before using Langchain and the REPL tool

**Reply by dvelm** [reply]
can you add more details for this ?

**Reply by patruff** [reply]
Sure, basically, just get Langchain and then you can ask

agent.run("What files are in my current directory?")

I have a video where I use GPT to create resources (a whole website) 

https://youtu.be/T3wWRd8KP5U

**Reply by patruff** [reply]
import os
from dotenv import load_dotenv

load_dotenv()

os.getenv('OPENAI_API_KEY')
os.getenv('SERPAPI_API_KEY')
os.getenv('STORAGE_BUCKET_NAME')

from langchain.agents import load_tools
from langchain.agents import initialize_agent
from langchain.llms import OpenAI

llm = OpenAI(temperature=0)
tools = load_tools(["serpapi", "llm-math","wikipedia","python_repl", "requests_all"], llm=llm)
agent = initialize_agent(tools, 
                         llm, 
                         agent="zero-shot-react-description", 
                         verbose=True)

**Reply by liteli1987gmail** [reply]
I try your code ,but pop that ValueError: Got unknown tool python_repl
